x-pack/plugins/apm/public/components/shared/dependencies_table/index.tsx

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiBasicTableColumn,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiInMemoryTable,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { EuiLink } from '@elastic/eui';
+import { ConnectionMetricItemWithComparisonData } from '../../../../common/connections';
+import {
+  asMillisecondDuration,
+  asPercent,
+  asTransactionRate,
+} from '../../../../common/utils/formatters';
+import { FETCH_STATUS } from '../../../hooks/use_fetcher';
+import { unit } from '../../../utils/style';
+import { SparkPlot } from '../charts/spark_plot';
+import { ImpactBar } from '../ImpactBar';
+import { ServiceMapLink } from '../Links/apm/ServiceMapLink';
+import { TableFetchWrapper } from '../table_fetch_wrapper';
+import { TruncateWithTooltip } from '../truncate_with_tooltip';
+import { OverviewTableContainer } from '../overview_table_container';
+
+export type DependenciesItem = Omit<
+  ConnectionMetricItemWithComparisonData,
+  'location'
+> & {
+  name: string;
+  href: string;
+  icon: React.ReactElement;
+};
+
+interface Props {
+  dependencies: DependenciesItem[];
+  serviceName?: string;
+  title: React.ReactNode;
+  nameColumnTitle: React.ReactNode;
+  status: FETCH_STATUS;
+  compact?: boolean;
+}
+
+export function DependenciesTable(props: Props) {
+  const {
+    dependencies,
+    serviceName,
+    title,
+    nameColumnTitle,
+    status,
+    compact = true,
+  } = props;
+
+  const pagination = compact
+    ? {
+        initialPageSize: 5,
+        pageSizeOptions: [5],
+        hidePerPageOptions: true,
+      }
+    : {};
+
+  const columns: Array<EuiBasicTableColumn<DependenciesItem>> = [
+    {
+      field: 'name',
+      name: nameColumnTitle,
+      render: (_, item) => {
+        const { name, href, icon } = item;
+        return (
+          <TruncateWithTooltip
+            text={name}
+            content={
+              <EuiFlexGroup gutterSize="s" responsive={false}>
+                <EuiFlexItem grow={false}>{icon}</EuiFlexItem>
```

Comment:
We might have to resolve conflicts between this and my changes in https://github.com/elastic/kibana/pull/107089/files#diff-732775f9fc7d8281a24dd8fef294675022aec9fab6f68546ebf75088304c83ffR123-R142